animation_nodes/nodes/generic/animate_data.py

```diff
@@ -39,3 +38,9 @@ def getExecutionCode(self, required):
         yield "influence = interpolation(influence)"
         yield getMixCode(self.dataType, "start", "end", "influence", "result")
         yield "outTime = time - finalDuration"
+
+def getMixCode(dataType, mix1 = "a", mix2 = "b", factor = "f", result = "result"):
+    if dataType in ("Float", "Vector", "Quaternion"): return f"{result} = {mix1} * (1 - {factor}) + {mix2} * {factor}"
+    if dataType == "Matrix": return f"{result} = {mix1}.lerp({mix2}, {factor})"
+    if dataType == "Color": return f"{result} = [v1 * (1 - {factor}) + v2 * {factor} for v1, v2 in zip({mix1}, {mix2})]"
+    if dataType == "Euler": return f"{result} = animation_nodes.utils.math.mixEulers({mix1}, {mix2}, {factor})"
```

Comment:
Keep this function in `mix_data`. But update the Color case, where we should create a color from the list. See below for how to incorporate it in the mix node.